api/v1beta1/azuremachine_default.go

```diff
@@ -67,7 +68,12 @@ func (s *AzureMachineSpec) SetDataDisksDefaults() {
 			}
 		}
 		if disk.CachingType == "" {
-			s.DataDisks[i].CachingType = "ReadWrite"
+			if s.DataDisks[i].ManagedDisk != nil &&
```

Comment:
I think we should be able to add one add'l test case to `TestAzureMachineSpec_SetDataDisksDefaults` in `api/v1beta1/azuremachine_default_test.go` to get UT coverage for this change.

Reply:
Oh yes, good catch!
Added now here: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2430/commits/69152d43c9c6f8bdb7560b8da8d5186b990c1b0c#diff-a4b7e2572f474ccf7af69f95fdde7fcda84526cca94cef837263765d9405d22eR218-R248